source/common/http/http1/codec_impl.cc

```diff
@@ -59,6 +59,10 @@ HeaderKeyFormatterPtr formatter(const Http::Http1Settings& settings) {
     return std::make_unique<ProperCaseHeaderKeyFormatter>();
   }
 
+  if (settings.header_key_format_ == Http1Settings::HeaderKeyFormat::Custom) {
+    return std::make_unique<CustomHeaderKeyFormatter>(settings.header_key_format_rules_);
+  }
+
```

Comment:
Now that there are >2 options, it probably makes sense to change this to a `switch`?